tesla_api/charge.py

```diff
@@ -1,24 +1,39 @@
+import asyncio
+
 class Charge:
     def __init__(self, api_client, vehicle_id):
         self._api_client = api_client
         self._vehicle_id = vehicle_id
 
-    def get_state(self):
-        return self._api_client.get('vehicles/{}/data_request/charge_state'.format(self._vehicle_id))
+    async def get_state(self):
+        return await self._api_client.get('vehicles/{}/data_request/charge_state'.format(self._vehicle_id))
 
-    def start_charging(self):
-        return self._api_client.post('vehicles/{}/command/charge_start'.format(self._vehicle_id))
+    async def start_charging(self):
+        return await self._api_client.post('vehicles/{}/command/charge_start'.format(self._vehicle_id))
 
-    def stop_charging(self):
-        return self._api_client.post('vehicles/{}/command/charge_stop'.format(self._vehicle_id))
+    async def stop_charging(self):
+        return await self._api_client.post('vehicles/{}/command/charge_stop'.format(self._vehicle_id))
 
-    def set_charge_limit(self, percentage):
+    async def set_charge_limit(self, percentage):
         percentage = round(percentage)
 
-        if percentage < 50 or percentage > 100:
+        if not (50 <= percentage <= 100):
             raise ValueError('Percentage should be between 50 and 100')
 
-        return self._api_client.post(
+        return await self._api_client.post(
             'vehicles/{}/command/set_charge_limit'.format(self._vehicle_id),
             {'percent': percentage}
-        )
\ No newline at end of file
+        )
+
+class ChargeSync(Charge):
```

Comment:
Is it conventional to create a synchronous wrapper called `xxxSync`? I saw you renamed the `TeslaApiClient` to have the `Async` suffix, so that seems to be the other way around?

Reply:
I used `TeslaApiClient` to ensure backwards compatibility with existing code, which is not needed for the individual classes (as they are all returned through the client).

However, I think for ease of development, ensuring that `Vehicle` and such refers to both classes (i.e. `VehicleSync` is a subclass of `Vehicle`), means that things that depend on type checks still work nicely. i.e. `isinstance(v, Vehicle)` or `def f(v: Vehicle):` will work correctly regardless of whether you use the sync or async client.

This also means old sync code that uses such type checks won't need to change those type checks if they want to switch to the async client.


p.s. I have another PR coming at the weekend for resuming the client from a previous token (in case you're thinking of making a new release with these changes).

Reply:
That makes sense. Now I'm not that familiar with the state of Python, but would you say it would be worth it to create a 2.0 release with only the async code? I mean, if everyone is moving to aiohttp instead of requests, then maybe we should just drop the sync stuff altogether?

Reply:
I'd be happy to move to async only, if you're not worried about maintaining backwards compatibility. I suppose people can always wrap the calls in `asyncio.get_event_loop().run_until_complete()` as I did in the Sync versions if they really need to run in a non-asyncio environment.

Reply:
I'm fine with moving to async only and releasing a 2.0 release. Could you do that?